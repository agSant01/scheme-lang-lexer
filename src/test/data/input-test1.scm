; arithmetic operations
(+ 2 #x4f)

; define/assign variables
(define var1 #b0)
(define foo 3)

; define funcs
(define (square x)
    (* x x))

; dummy fun for testing booleans
(define (trueOrFalse val)
    (if val #t #f)
)

;; using the func
(square 4) ; output: 4^2 = 16
(+ (square 2) (square 3)) ; output: 2^2 + 3^2 = 4 + 9 = 13

; flow control
(define (abs x)
    (if (< x 0) 
        (- x) ; if
        x     ; else
    ))
;; calling abs
(format "AbsFunc result: ~s=~a" (abs -4) (abs -4))

; list
(define boo (list 9 3 5))
(println boo)
(sort boo <) ; output: (4 5 9)
(length (list 3 #o4)) ; output: 2

; code and data
(define x (list 1 2 3)) 
(car x) ; gives the first element
(cdr x) ; gives back the other things

(println "")
(println "Cons examples")
(cons "a" "b")
(cons (cons 5 6) 7) ; creates pairs
(define test (list 3 4 5))

; testing comlex numbers
(define complex1 +4-5i)

(define complex3 -24+2i)

; testing the complex numbers of diferenc bases
(define complexOctal #o-24+2i)
(define complexD #d-24+2i)
(define complexBin #b010-001i)
(define complexHex #x-24+2i)

(define float 2.0)
(define signedFloat -2.0)

(define <yupThisIsAnIdentifier>> "Incluiding the '>' and '>>'")

(define decWithExpPositive 42462e-32)
(define decWithExpNegative -42462e-32)

(#o4 "item2" #b0011100 #x32fa)

; testing vectors
#(#f #t 78)

; abbreviation test
(define lisstOfAbbs (list '* `3523 '))

`(list ,(+ 1 2) 4)
(let ((name 'a)) `(list ,name ',name))
`#(10 5 ,(sqrt 4) ,@(map sqrt '(16 9)) 8)
