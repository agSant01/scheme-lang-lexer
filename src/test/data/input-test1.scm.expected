TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:1; COLUMN:0]
TOKEN [TYPE:SIGN; TEXT:'+'; LINE:1; COLUMN:1]
TOKEN [TYPE:UINTEGER10; TEXT:'2'; LINE:1; COLUMN:3]
TOKEN [TYPE:NUM16; TEXT:'#x4f'; LINE:1; COLUMN:5]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:1; COLUMN:9]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:4; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:4; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'var1'; LINE:4; COLUMN:8]
TOKEN [TYPE:NUM2; TEXT:'#b0'; LINE:4; COLUMN:13]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:4; COLUMN:16]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:5; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:5; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'foo'; LINE:5; COLUMN:8]
TOKEN [TYPE:UINTEGER10; TEXT:'3'; LINE:5; COLUMN:12]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:5; COLUMN:13]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:8; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:8; COLUMN:1]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:8; COLUMN:8]
TOKEN [TYPE:IDENTIFIER; TEXT:'square'; LINE:8; COLUMN:9]
TOKEN [TYPE:IDENTIFIER; TEXT:'x'; LINE:8; COLUMN:16]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:8; COLUMN:17]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:9; COLUMN:4]
TOKEN [TYPE:IDENTIFIER; TEXT:'*'; LINE:9; COLUMN:5]
TOKEN [TYPE:IDENTIFIER; TEXT:'x'; LINE:9; COLUMN:7]
TOKEN [TYPE:IDENTIFIER; TEXT:'x'; LINE:9; COLUMN:9]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:9; COLUMN:10]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:9; COLUMN:11]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:12; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:12; COLUMN:1]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:12; COLUMN:8]
TOKEN [TYPE:IDENTIFIER; TEXT:'trueOrFalse'; LINE:12; COLUMN:9]
TOKEN [TYPE:IDENTIFIER; TEXT:'val'; LINE:12; COLUMN:21]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:12; COLUMN:24]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:13; COLUMN:4]
TOKEN [TYPE:KEYWORD; TEXT:'if'; LINE:13; COLUMN:5]
TOKEN [TYPE:IDENTIFIER; TEXT:'val'; LINE:13; COLUMN:8]
TOKEN [TYPE:BOOLEAN; TEXT:'#t'; LINE:13; COLUMN:12]
TOKEN [TYPE:BOOLEAN; TEXT:'#f'; LINE:13; COLUMN:15]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:13; COLUMN:17]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:14; COLUMN:0]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:17; COLUMN:0]
TOKEN [TYPE:IDENTIFIER; TEXT:'square'; LINE:17; COLUMN:1]
TOKEN [TYPE:UINTEGER10; TEXT:'4'; LINE:17; COLUMN:8]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:17; COLUMN:9]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:18; COLUMN:0]
TOKEN [TYPE:SIGN; TEXT:'+'; LINE:18; COLUMN:1]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:18; COLUMN:3]
TOKEN [TYPE:IDENTIFIER; TEXT:'square'; LINE:18; COLUMN:4]
TOKEN [TYPE:UINTEGER10; TEXT:'2'; LINE:18; COLUMN:11]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:18; COLUMN:12]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:18; COLUMN:14]
TOKEN [TYPE:IDENTIFIER; TEXT:'square'; LINE:18; COLUMN:15]
TOKEN [TYPE:UINTEGER10; TEXT:'3'; LINE:18; COLUMN:22]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:18; COLUMN:23]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:18; COLUMN:24]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:21; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:21; COLUMN:1]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:21; COLUMN:8]
TOKEN [TYPE:KEYWORD; TEXT:'abs'; LINE:21; COLUMN:9]
TOKEN [TYPE:IDENTIFIER; TEXT:'x'; LINE:21; COLUMN:13]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:21; COLUMN:14]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:22; COLUMN:4]
TOKEN [TYPE:KEYWORD; TEXT:'if'; LINE:22; COLUMN:5]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:22; COLUMN:8]
TOKEN [TYPE:IDENTIFIER; TEXT:'<'; LINE:22; COLUMN:9]
TOKEN [TYPE:IDENTIFIER; TEXT:'x'; LINE:22; COLUMN:11]
TOKEN [TYPE:UINTEGER10; TEXT:'0'; LINE:22; COLUMN:13]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:22; COLUMN:14]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:23; COLUMN:8]
TOKEN [TYPE:SIGN; TEXT:'-'; LINE:23; COLUMN:9]
TOKEN [TYPE:IDENTIFIER; TEXT:'x'; LINE:23; COLUMN:11]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:23; COLUMN:12]
TOKEN [TYPE:IDENTIFIER; TEXT:'x'; LINE:24; COLUMN:8]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:25; COLUMN:4]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:25; COLUMN:5]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:27; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'format'; LINE:27; COLUMN:1]
TOKEN [TYPE:STRING; TEXT:'"AbsFunc result: ~s=~a"'; LINE:27; COLUMN:8]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:27; COLUMN:32]
TOKEN [TYPE:KEYWORD; TEXT:'abs'; LINE:27; COLUMN:33]
TOKEN [TYPE:REAL10; TEXT:'-4'; LINE:27; COLUMN:37]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:27; COLUMN:39]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:27; COLUMN:41]
TOKEN [TYPE:KEYWORD; TEXT:'abs'; LINE:27; COLUMN:42]
TOKEN [TYPE:REAL10; TEXT:'-4'; LINE:27; COLUMN:46]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:27; COLUMN:48]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:27; COLUMN:49]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:30; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:30; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'boo'; LINE:30; COLUMN:8]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:30; COLUMN:12]
TOKEN [TYPE:KEYWORD; TEXT:'list'; LINE:30; COLUMN:13]
TOKEN [TYPE:UINTEGER10; TEXT:'9'; LINE:30; COLUMN:18]
TOKEN [TYPE:UINTEGER10; TEXT:'3'; LINE:30; COLUMN:20]
TOKEN [TYPE:UINTEGER10; TEXT:'5'; LINE:30; COLUMN:22]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:30; COLUMN:23]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:30; COLUMN:24]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:31; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'println'; LINE:31; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'boo'; LINE:31; COLUMN:9]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:31; COLUMN:12]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:32; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'sort'; LINE:32; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'boo'; LINE:32; COLUMN:6]
TOKEN [TYPE:IDENTIFIER; TEXT:'<'; LINE:32; COLUMN:10]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:32; COLUMN:11]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:33; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'length'; LINE:33; COLUMN:1]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:33; COLUMN:8]
TOKEN [TYPE:KEYWORD; TEXT:'list'; LINE:33; COLUMN:9]
TOKEN [TYPE:UINTEGER10; TEXT:'3'; LINE:33; COLUMN:14]
TOKEN [TYPE:NUM8; TEXT:'#o4'; LINE:33; COLUMN:16]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:33; COLUMN:19]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:33; COLUMN:20]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:36; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:36; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'x'; LINE:36; COLUMN:8]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:36; COLUMN:10]
TOKEN [TYPE:KEYWORD; TEXT:'list'; LINE:36; COLUMN:11]
TOKEN [TYPE:UINTEGER10; TEXT:'1'; LINE:36; COLUMN:16]
TOKEN [TYPE:UINTEGER10; TEXT:'2'; LINE:36; COLUMN:18]
TOKEN [TYPE:UINTEGER10; TEXT:'3'; LINE:36; COLUMN:20]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:36; COLUMN:21]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:36; COLUMN:22]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:37; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'car'; LINE:37; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'x'; LINE:37; COLUMN:5]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:37; COLUMN:6]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:38; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'cdr'; LINE:38; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'x'; LINE:38; COLUMN:5]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:38; COLUMN:6]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:40; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'println'; LINE:40; COLUMN:1]
TOKEN [TYPE:STRING; TEXT:'""'; LINE:40; COLUMN:9]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:40; COLUMN:11]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:41; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'println'; LINE:41; COLUMN:1]
TOKEN [TYPE:STRING; TEXT:'"Cons examples"'; LINE:41; COLUMN:9]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:41; COLUMN:24]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:42; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'cons'; LINE:42; COLUMN:1]
TOKEN [TYPE:STRING; TEXT:'"a"'; LINE:42; COLUMN:6]
TOKEN [TYPE:STRING; TEXT:'"b"'; LINE:42; COLUMN:10]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:42; COLUMN:13]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:43; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'cons'; LINE:43; COLUMN:1]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:43; COLUMN:6]
TOKEN [TYPE:KEYWORD; TEXT:'cons'; LINE:43; COLUMN:7]
TOKEN [TYPE:UINTEGER10; TEXT:'5'; LINE:43; COLUMN:12]
TOKEN [TYPE:UINTEGER10; TEXT:'6'; LINE:43; COLUMN:14]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:43; COLUMN:15]
TOKEN [TYPE:UINTEGER10; TEXT:'7'; LINE:43; COLUMN:17]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:43; COLUMN:18]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:44; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:44; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'test'; LINE:44; COLUMN:8]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:44; COLUMN:13]
TOKEN [TYPE:KEYWORD; TEXT:'list'; LINE:44; COLUMN:14]
TOKEN [TYPE:UINTEGER10; TEXT:'3'; LINE:44; COLUMN:19]
TOKEN [TYPE:UINTEGER10; TEXT:'4'; LINE:44; COLUMN:21]
TOKEN [TYPE:UINTEGER10; TEXT:'5'; LINE:44; COLUMN:23]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:44; COLUMN:24]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:44; COLUMN:25]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:47; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:47; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'complex1'; LINE:47; COLUMN:8]
TOKEN [TYPE:COMPLEX10; TEXT:'+4-5i'; LINE:47; COLUMN:17]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:47; COLUMN:22]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:49; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:49; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'complex3'; LINE:49; COLUMN:8]
TOKEN [TYPE:COMPLEX10; TEXT:'-24+2i'; LINE:49; COLUMN:17]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:49; COLUMN:23]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:52; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:52; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'complexOctal'; LINE:52; COLUMN:8]
TOKEN [TYPE:NUM8; TEXT:'#o-24+2i'; LINE:52; COLUMN:21]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:52; COLUMN:29]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:53; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:53; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'complexD'; LINE:53; COLUMN:8]
TOKEN [TYPE:NUM10; TEXT:'#d-24+2i'; LINE:53; COLUMN:17]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:53; COLUMN:25]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:54; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:54; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'complexBin'; LINE:54; COLUMN:8]
TOKEN [TYPE:NUM2; TEXT:'#b010-001i'; LINE:54; COLUMN:19]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:54; COLUMN:29]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:55; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:55; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'complexHex'; LINE:55; COLUMN:8]
TOKEN [TYPE:NUM16; TEXT:'#x-24+2i'; LINE:55; COLUMN:19]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:55; COLUMN:27]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:57; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:57; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'float'; LINE:57; COLUMN:8]
TOKEN [TYPE:UREAL10; TEXT:'2.0'; LINE:57; COLUMN:14]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:57; COLUMN:17]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:58; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:58; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'signedFloat'; LINE:58; COLUMN:8]
TOKEN [TYPE:REAL10; TEXT:'-2.0'; LINE:58; COLUMN:20]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:58; COLUMN:24]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:60; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:60; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'<yupThisIsAnIdentifier>>'; LINE:60; COLUMN:8]
TOKEN [TYPE:STRING; TEXT:'"Incluiding the '>' and '>>'"'; LINE:60; COLUMN:33]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:60; COLUMN:62]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:62; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:62; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'decWithExpPositive'; LINE:62; COLUMN:8]
TOKEN [TYPE:UREAL10; TEXT:'42462e-32'; LINE:62; COLUMN:27]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:62; COLUMN:36]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:63; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:63; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'decWithExpNegative'; LINE:63; COLUMN:8]
TOKEN [TYPE:REAL10; TEXT:'-42462e-32'; LINE:63; COLUMN:27]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:63; COLUMN:37]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:65; COLUMN:0]
TOKEN [TYPE:NUM8; TEXT:'#o4'; LINE:65; COLUMN:1]
TOKEN [TYPE:STRING; TEXT:'"item2"'; LINE:65; COLUMN:5]
TOKEN [TYPE:NUM2; TEXT:'#b0011100'; LINE:65; COLUMN:13]
TOKEN [TYPE:NUM16; TEXT:'#x32fa'; LINE:65; COLUMN:23]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:65; COLUMN:29]
TOKEN [TYPE:OPEN_VECTOR; TEXT:'#('; LINE:68; COLUMN:0]
TOKEN [TYPE:BOOLEAN; TEXT:'#f'; LINE:68; COLUMN:2]
TOKEN [TYPE:BOOLEAN; TEXT:'#t'; LINE:68; COLUMN:5]
TOKEN [TYPE:UINTEGER10; TEXT:'78'; LINE:68; COLUMN:8]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:68; COLUMN:10]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:71; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'define'; LINE:71; COLUMN:1]
TOKEN [TYPE:IDENTIFIER; TEXT:'lisstOfAbbs'; LINE:71; COLUMN:8]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:71; COLUMN:20]
TOKEN [TYPE:KEYWORD; TEXT:'list'; LINE:71; COLUMN:21]
TOKEN [TYPE:ABBREVIATION; TEXT:''*'; LINE:71; COLUMN:26]
TOKEN [TYPE:ABBREVIATION; TEXT:'`3523'; LINE:71; COLUMN:29]
TOKEN [TYPE:QUOTATION_MARK; TEXT:'''; LINE:71; COLUMN:35]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:71; COLUMN:36]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:71; COLUMN:37]
TOKEN [TYPE:QUOTATION_MARK; TEXT:'`'; LINE:73; COLUMN:0]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:73; COLUMN:1]
TOKEN [TYPE:KEYWORD; TEXT:'list'; LINE:73; COLUMN:2]
TOKEN [TYPE:UNQUOTATION_MARK; TEXT:','; LINE:73; COLUMN:7]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:73; COLUMN:8]
TOKEN [TYPE:SIGN; TEXT:'+'; LINE:73; COLUMN:9]
TOKEN [TYPE:UINTEGER10; TEXT:'1'; LINE:73; COLUMN:11]
TOKEN [TYPE:UINTEGER10; TEXT:'2'; LINE:73; COLUMN:13]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:73; COLUMN:14]
TOKEN [TYPE:UINTEGER10; TEXT:'4'; LINE:73; COLUMN:16]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:73; COLUMN:17]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:74; COLUMN:0]
TOKEN [TYPE:KEYWORD; TEXT:'let'; LINE:74; COLUMN:1]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:74; COLUMN:5]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:74; COLUMN:6]
TOKEN [TYPE:IDENTIFIER; TEXT:'name'; LINE:74; COLUMN:7]
TOKEN [TYPE:ABBREVIATION; TEXT:''a'; LINE:74; COLUMN:12]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:74; COLUMN:14]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:74; COLUMN:15]
TOKEN [TYPE:QUOTATION_MARK; TEXT:'`'; LINE:74; COLUMN:17]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:74; COLUMN:18]
TOKEN [TYPE:KEYWORD; TEXT:'list'; LINE:74; COLUMN:19]
TOKEN [TYPE:ABBREVIATION; TEXT:',name'; LINE:74; COLUMN:24]
TOKEN [TYPE:QUOTATION_MARK; TEXT:'''; LINE:74; COLUMN:30]
TOKEN [TYPE:ABBREVIATION; TEXT:',name'; LINE:74; COLUMN:31]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:74; COLUMN:36]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:74; COLUMN:37]
TOKEN [TYPE:QUOTATION_MARK; TEXT:'`'; LINE:75; COLUMN:0]
TOKEN [TYPE:OPEN_VECTOR; TEXT:'#('; LINE:75; COLUMN:1]
TOKEN [TYPE:UINTEGER10; TEXT:'10'; LINE:75; COLUMN:3]
TOKEN [TYPE:UINTEGER10; TEXT:'5'; LINE:75; COLUMN:6]
TOKEN [TYPE:UNQUOTATION_MARK; TEXT:','; LINE:75; COLUMN:8]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:75; COLUMN:9]
TOKEN [TYPE:IDENTIFIER; TEXT:'sqrt'; LINE:75; COLUMN:10]
TOKEN [TYPE:UINTEGER10; TEXT:'4'; LINE:75; COLUMN:15]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:75; COLUMN:16]
TOKEN [TYPE:UNQUOTATION_MARK; TEXT:',@'; LINE:75; COLUMN:18]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:75; COLUMN:20]
TOKEN [TYPE:IDENTIFIER; TEXT:'map'; LINE:75; COLUMN:21]
TOKEN [TYPE:IDENTIFIER; TEXT:'sqrt'; LINE:75; COLUMN:25]
TOKEN [TYPE:QUOTATION_MARK; TEXT:'''; LINE:75; COLUMN:30]
TOKEN [TYPE:OPEN_PAREN; TEXT:'('; LINE:75; COLUMN:31]
TOKEN [TYPE:UINTEGER10; TEXT:'16'; LINE:75; COLUMN:32]
TOKEN [TYPE:UINTEGER10; TEXT:'9'; LINE:75; COLUMN:35]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:75; COLUMN:36]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:75; COLUMN:37]
TOKEN [TYPE:UINTEGER10; TEXT:'8'; LINE:75; COLUMN:39]
TOKEN [TYPE:CLOSE_PAREN; TEXT:')'; LINE:75; COLUMN:40]
